sources/packages/web/src/views/aest/institution/ViewOfferingChangeRequest.vue

```diff
@@ -109,17 +101,27 @@ export default defineComponent({
   setup(props) {
     const tab = ref("requested-change");
     const headerDetails = ref({} as ProgramOfferingHeader);
+    const offering = ref({} as EducationProgramOfferingAPIOutDTO);
     const assessOfferingChangeModalRef = ref(
       {} as ModalDialog<OfferingChangeAssessmentAPIInDTO | boolean>,
     );
     const snackBar = useSnackBar();
     const router = useRouter();
 
-    //TODO: This callback implementation needs to be removed when the program and offering header component
-    //TODO: is enhanced to load header values with it's own API call.
-    const getHeaderDetails = (data: ProgramOfferingHeader) => {
-      headerDetails.value = data;
-    };
+    onMounted(async () => {
+      offering.value =
+        await EducationProgramOfferingService.shared.getOfferingDetails(
+          props.offeringId,
+        );
+      headerDetails.value = {
+        institutionName: offering.value.institutionName,
+        submittedDate: offering.value.submittedDate,
+        status: offering.value.offeringStatus,
+        assessedBy: offering.value.assessedBy,
+        assessedDate: offering.value.assessedDate,
+        locationName: offering.value.locationName,
+      };
+    });
```

Comment:
just thinking, should this logic be part of a watch as `props.offeringId,` can be changed?

Reply:
From my understanding, whenever `offeringId` changes, it will be when the entire view refreshes and `onMount` will be called again in that scenario. Please let me know if I am missing something.